src/libraries/System.Net.Http/src/System/Net/Http/HttpMessageInvoker.cs

```diff
@@ -69,47 +105,55 @@ public virtual Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, C
 
             ObjectDisposedException.ThrowIf(_disposed, this);
 
+            EnsureMetrics();
+
             if (ShouldSendWithTelemetry(request))
             {
-                return SendAsyncWithTelemetry(_handler, request, cancellationToken);
+                return SendAsyncWithTelemetry(_handler, request, _metrics, cancellationToken);
             }
 
             return _handler.SendAsync(request, cancellationToken);
 
-            static async Task<HttpResponseMessage> SendAsyncWithTelemetry(HttpMessageHandler handler, HttpRequestMessage request, CancellationToken cancellationToken)
+            static async Task<HttpResponseMessage> SendAsyncWithTelemetry(HttpMessageHandler handler, HttpRequestMessage request, HttpMetrics metrics, CancellationToken cancellationToken)
             {
+                long startTimestamp = Stopwatch.GetTimestamp();
+
                 HttpTelemetry.Log.RequestStart(request);
+                metrics.RequestStart(request);
 
                 HttpResponseMessage? response = null;
+                Exception? unhandledException = null;
                 try
                 {
                     response = await handler.SendAsync(request, cancellationToken).ConfigureAwait(false);
                     return response;
                 }
-                catch (Exception ex) when (LogRequestFailed(ex, telemetryStarted: true))
+                catch (Exception ex) when (LogRequestFailed(ex, telemetryStarted: true, out unhandledException))
```

Comment:
I don't know why logging happens here instead of the catch block. Capturing the unhandled exception would be easier if this logic was in the catch.

Reply:
I believe it's to capture the logging scopes before unwinding the stack.